rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -771,7 +769,46 @@ type StrikeSpread: <"A class for defining a strike spread feature.">
     upperStrike OptionStrike (1..1) <"Upper strike in a strike spread.">
     upperStrikeNumberOfOptions number (1..1) <"Number of options at the upper strike price in a strike spread.">
 
+type Exercise: <"A class defining the exercise period for an option together with any rules governing the notional amount of the underlying which can be exercised on any given exercise date and any associated exercise fees.">
+	[metadata key]
+
+    style OptionExerciseStyleEnum (1..1) <"Whether the option has a single exercise (european), multiple exercise dates (bermuda), or a continuous range of exercise (american).">
+	commencementDate AdjustableOrRelativeDate (0..1) <"The first day of the exercise period for an American style option.">
+    exerciseDates AdjustableOrRelativeDates (0..1) <"The dates that define the Bermuda option exercise dates and the expiration date. The last specified date is assumed to be the expiration date. The dates can either be specified as a series of explicit dates and associated adjustments or as a series of dates defined relative to another schedule of dates, for example, the calculation period start dates. Where a relative series of dates are defined the first and last possible exercise dates can be separately specified.">
```

Comment:
Cardinality of `exerciseDates` (`(0..1)`) vs `expirationDate` (`(0..*)`) look incorrectly swapped.

Reply:
This is actually in line with what was in `bermudaExercise` and `european/americanExercise`. It is okay I think because `exerciseDates` is of type "_AdjustableOrRelativeDates_", which contains `adjustableDates -> unadjustedDate/adjusteDate` which are (0..*) cardinality (also, `exerciseDates` was also (0..1) cardinality in bermudaExercise). As for `expirationDate`, it was multiple cardinality in `europeanExercise`, so I've kept it like this in here to keep the model "loose" and not break some samples.

Reply:
I think the reason why it was multiple cardinality is that in FpML, commodity options allow for mutliple expirationDate, and if we constrict the cardinality to `(0..1)` it will break some of the mappings. Specifically, these 2 FpML samples have 3 `expirationDate` fields within `europeanExercise`, which all currently map to `exerciseTerms` in CDM, but if we change it to (0..1) only one of the 3 dates will map:

- com ex08 oil call option strip
- com ex41 oil asian barrier option strip

![image](https://github.com/finos/common-domain-model/assets/141412602/357311d2-a335-4daa-ba36-5e82c7dea616)

Reply:
The sample name refers to "Strip". I suspect this is one of 2 cases:
* Either this is actually a "strip" of options, i.e. effectively 3 distinct options with same characteristics but 3 separate exercise dates (like a a cap is a strip of caplets) --> under the current design this should be 3 separate "European" options
* Or it's a single option that may be exercised on 3 dates --> in this case it's a "Bermuda"

Qn: was the cardinality of `expirationDate` extended to `(1..*)` specifically to accomodate the Commodity use-case, or was it always like that before the Commodity work?

Again, I think this one merits a further discussion.

Reply:
I think probably the first case you mention of the strip of options is the most likely one , as indicated by the sample names and the existence of multiple cardinality in FpML for commodity strips:

![image](https://github.com/finos/common-domain-model/assets/141412602/0945774b-467a-4e30-bbd5-3a32390e0b86)

I am uncertain if the introduction of the multiple cardinality in CDM was intended to allow for these commodity use-cases, and whether the presence of these 3 `expirationDate` should result in the creation of 3 CDM objects or not. However, I don't think they should be split given that the commodity strip likely represents only one trade/product. I've discussed this with @mgratacos and we've agreed in the meantime to seek clarification from Caroline at BNPP given her extensive experience with commodities, and to also raise this in the next DPBE WG call.

Reply:
Once this cardinality conundrum has been resolved I think this PR looks good to go 👍

Reply:
Hi @chrisisla ,

This has recently been discussed with Caroline, and it's been cleared up that in the case for commodity asian options with a strip of periods, they need the `expirationDate` field to be **multiple** cardinality. Essentially, these type of products contain a "strip" of periods each with an exercise date, but the trade itself is booked and processed as **one** single option. In each of these periods, the average of the floating rate is calculated to be compared against the strike in each of the periods. Then the payoff is determined to be in or out of the money (with respect to the strike), and if in the money the cash payment is paid on each of the relevant settlement dates of the strip periods.

@lolabeis Please do let me know if I missed any relevant information in this description.

Note this PR has been closed due to conflicts - a new one has been generated: #2687